docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,112 @@
+# Storage Node Downtime Disqualification
+
+## Abstract
+
+This document describes storage node downtime suspension and disqualification.
+
+## Background
+
+The previous implementation of downtime disqualification led to the unfair disqualification of storage nodes. This prompted a halt to downtime disqualification and a new design for [Downtime Tracking](storage-node-downtime-tracking.md).
+
+While the previously mentioned blueprint presents a means of tracking downtime, it leaves out the design of how to use this information for disqualification. That is the purpose of this blueprint.
+
+## Design
+
+### Suspension
+
+Storage node downtime can have a range of causes. For those storage node operators who may have fallen victim to a temporary issue, we want to give them a chance to diagnose and fix it before disqualifying them for good. For this reason, we are introducing downtime suspension as a component of disqualification.
+
+For downtime suspension and disqualification we need a few new configurable values:
+
+- Tracking Period: The length of time into the past for which we measure downtime.
+- Allowed Downtime: The amount of downtime within the tracking period we will allow before suspension.
+- Grace Period: The length of time to give suspended SNOs to diagnose and fix issues causing downtime. Afterwards, they will have one tracking period to demonstrate an acceptable amount of downtime before disqualification.
+
+When the [downtime tracking system](storage-node-downtime-tracking.md) adds an entry to the nodes_offline_time table, if the node is not already suspended, we check the total amount of downtime for the tracking period, e.g. if the tracking period is 30 days, we sum downtime for the last 30 days. If the total is greater than the allowed downtime, the node is suspended and put into an "under review" state for a length of time equal to the grace period + one tracking period.
+
+Suspended nodes are not selected for uploads, and all pieces they hold are considered unhealthy. This means that we consider these pieces to be unreliable. If the number of healthy pieces for a segment drops below the repair threshold, the segment will be repaired and some, or all, of the unhealthy pieces may be moved onto other healthy nodes. Nodes which previously held these pieces which were "repaired" should eventually be told to put them into the trash by the garbage collection service. Until the segment falls into repair and the piece is given to another node, it is still possible to receive download requests for this piece while in suspension mode.
```

Comment:
Does node in `under_review` get selected for uploads? If so, they can get new pieces and then get paid for storing those pieces, go offline, we put them into suspension mode which causes more repair traffic for those new pieces?

Reply:
Added to Open Issues section of new document